### DISCUSSION TITLE
Great Work!

### DISCUSSION THREAD

**Discussion by xKean** [AIrjen/OneButtonPrompt#19] [Ideas]
Hi,
I have been looking for exactly this for the last 3 months! 😄 
I forked your repo and added some custom code!

- Automatic Quality Gate
- Random Model Selection
- Random Negative Embeddings usage
- Different Upscaling
- WebuiApi from @mix1009 (THANKS!)

If you'd like to implement some of those ideas, you can watch my fork and get an idea of how I did it.
The code I wrote is not ready for a PR but I could polish it further if you'd like.
Otherwise, I only say THANK YOU!

Keep it up! 😎

**Comment by AIrjen** [comment]
Hi XKean, this looks very interesting.
The running automatic part is a personal part of this project. Actually that was just what it was, before I converted it to be used in A1111. Most people don't even know its there. I'm just lazy and left it in there, because then I won't have to manage multiple projects. I still use it for testing a lot, or creating images while I'm doing my normal job.

Also don't be scared about your PR's. This is my first python project, so to a professional developer, my code is probably not the best anyway.

I think these are solid additions. I love the idea with the integration as an 'Automatic Quality Gate'. Thats a new idea for me, and I am very curious about the results!

The random models etc, all make sense. If I didn't spend time converting it to A1111, I'd probably had done the same.

I'm all open for a community driven project, so we could bring the things in.

The only things that come to mind are:
- I'm not sure of bringing the entire copy of the Aesthetic Score in this repo. Is it possible to detect a normal install? I'm not sure what is best here
- Would be good to: make your model, upscalers and negative lists into a csv, and place then in a new directory. So other users can easily change and add their own stuff.

Basically, work it out to be more of an official feature, instead of a personal project.

In short, thanks for this. Very inspiring, and you have given me a lot to think about!

**Comment by AIrjen** [comment]
So, I had some time this weekend away from the computer to think about this some more.

The main problem is that the code for the automatic running and upscaling runs completely seperate currently from the A1111 extension. While both use the prompt generator logic, they are now seperate parts of the whole.

So this is what I'm thinking. Create an interface (new tab) in A1111 as part of the One Buttom Prompt extension, that can manage and run this part. Meaning that is should be affected by the other settings set in the main or advanced tab.
This should also mean, that here it should be possible to select your model/upscaler/strength's etc.

This way, both things will be moved under a single method of working, and we then have an interface for a regular user to work with.

This is then something of a long term thing, since it requires a refactoring of some current code for it to work.
I'd also like some of the fields to be dynamic (such as model, upscalers incl random) and I have no idea how to do that yet.
I still also really like the idea of integrating the automatic quality gate in this part.

all in all, this will be the next big thing to add!

(ps, still need a good name for this)
